src/App.tsx

```diff
@@ -15,36 +18,110 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortBy {
+  Default,
+  Name,
+  Length,
+}
+
+type SortTheGoods = (
+  items: Good[],
+  sortBy: SortBy | null,
+  option: boolean,
+) => Good[];
+
+const getSortedGoods: SortTheGoods = (goods, sortBy, isReverseAdded) => {
+  const sortedGoods: Good[] = [...goods];
+
+  if (sortBy) {
+    sortedGoods.sort((good1, good2) => {
+      switch (sortBy) {
+        case SortBy.Name:
+          return good1.localeCompare(good2);
+
+        case SortBy.Length:
+          return good1.length - good2.length;
+
+        default:
+          return SortBy.Default;
+      }
+    });
+  }
+
+  if (isReverseAdded) {
+    return sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+};
+
 export const App: React.FC = () => {
+  const [sortBy, setSortBy] = useState<SortBy | null>(null);
+  const [isReverseAdded, setIsReverseAdded] = useState(false);
+
+  const visibleGoods: Good[] = getSortedGoods(
+    goodsFromServer,
+    sortBy,
+    isReverseAdded,
+  );
+
+  const isResetButtonVisible = isReverseAdded || sortBy;
+
+  const resetSort = () => {
+    setSortBy(null);
+    setIsReverseAdded(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortBy !== SortBy.Name,
+          })}
+          onClick={() => setSortBy(SortBy.Name)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
```

Comment:
This had is-success class instead of is-info

Reply:
> This had is-success class instead of is-info

Good point, the color is different in demo